crates/rattler_solve/src/libsolv/wrapper/solve_goal.rs

```diff
@@ -0,0 +1,106 @@
+use super::ffi;
+use crate::libsolv::wrapper::ffi::{
+    SOLVER_DISFAVOR, SOLVER_ERASE, SOLVER_FAVOR, SOLVER_INSTALL, SOLVER_LOCK, SOLVER_SOLVABLE,
+    SOLVER_SOLVABLE_PROVIDES, SOLVER_UPDATE, SOLVER_WEAK,
+};
+use crate::libsolv::wrapper::pool::MatchSpecId;
+use crate::libsolv::wrapper::solvable::SolvableId;
+use std::os::raw::c_int;
+
+/// Wrapper for libsolv queue type that stores jobs for the solver. This type provides a safe
+/// wrapper around a goal state for libsolv.
+pub struct SolveGoal {
+    queue: ffi::Queue,
```

Comment:
Why not use the existing `Queue<T>` wrapper here (i.e. `queue: Queue<SolvableId>`)? I think we are currently duplicating part of that code (which seems extra undesirable because it is unsafe)

Reply:
Because the SolverGoal needs to store both `SolvableId` and `MatchSpecId` in the queue. I could make an enum out of that and use `Queue<SolvableOrMatchSpecId>`. WDYT?